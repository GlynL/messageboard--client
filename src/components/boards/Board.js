import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../../actions/boards";
import Threads from "./Threads";

class Board extends Component {
  componentDidMount() {
    this.props.fetchBoards();
  }

  componentDidUpdate() {
    this.props.setActive(this.props.boards, this.props.match.params.board);
  }

  render() {
    if (
      Object.keys(this.props.board).length === 0 &&
      this.props.board.constructor === Object
    ) {
      return <div>loading...</div>;
    }
    return (
      <div>
        <h1>{this.props.board.name}</h1>
        <Threads />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  board: state.boards.activeBoard,
  boards: state.boards.boards
});

export default connect(
  mapStateToProps,
  actions
)(Board);
