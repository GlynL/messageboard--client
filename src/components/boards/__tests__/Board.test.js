import React from "react";
import { shallow } from "enzyme";
import { Board } from "../Board";

describe("with board prop -loaded from componentDidMount", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Board
        match={{ params: { board: "test board" } }}
        board={{ name: "test board" }}
        setActive={jest.fn()}
      />
    );
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("without board fetched", () => {
  let wrapper, setActive;
  beforeEach(() => {
    setActive = jest.fn();
    wrapper = shallow(
      <Board
        match={{ params: { board: "test board" } }}
        setActive={setActive}
        board={{}}
      />
    );
  });
  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calls setActive in componentDidMount", () => {
    expect(setActive).toHaveBeenCalledWith(
      wrapper.instance().props.match.params.board
    );
  });
});
