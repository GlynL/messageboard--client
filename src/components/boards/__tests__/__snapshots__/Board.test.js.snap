// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with board prop -loaded from componentDidMount matches snapshot 1`] = `
<section>
  <h1
    className="board__title"
  >
    test board
  </h1>
  <withRouter(Connect(Threads)) />
</section>
`;

exports[`without board fetched matches snapshot 1`] = `
<div>
  loading...
</div>
`;
